src/MiniExcel/DbList.cs

```diff
@@ -0,0 +1,176 @@
+        {
+            _cmd.CommandText = $"SELECT MAX(`index`) FROM {_tableName}";
+            var result = _cmd.ExecuteScalar();
+            if (result == DBNull.Value)
+                return -1;
+
+            return (long)result;
+        }
+
+        public void Clear()
+        {
+            _cmd.CommandText = $"DELETE FROM {_tableName}";
+            _cmd.ExecuteNonQuery();
+            Count = 0;
+        }
+
+        public bool Contains(string item)
+        {
+            _cmd.CommandText = $"SELECT * FROM {_tableName} WHERE name = '{item}'";
+            return _cmd.ExecuteScalar() != null;
+        }
+
+        public void CopyTo(string[] array, int arrayIndex)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void AddRange(List<string> array)
+        {
+            var maxIndex = GetMaxIndex();
+
+            var cmdTxt = new StringBuilder();
+
+            cmdTxt.Append($"INSERT INTO {_tableName}(name, `index`) VALUES");
+            for (var i = 0; i < array.Count; i++)
+            {
+                var item = array[i];
+                cmdTxt.Append($"('{item}', {maxIndex + i + 1})");
+                cmdTxt.Append(i != array.Count - 1 ? ',' : ';');
+            }
+
+            _cmd.CommandText = cmdTxt.ToString();
+            _cmd.ExecuteNonQuery();
+            Count += array.Count;
+        }
+
+        public bool Remove(string item)
+        {
+            var index = IndexOf(item);
+            RemoveAt(index);
+            return true;
+        }
+
+        public int Count { get; private set; }
+
+        public bool IsReadOnly { get; }
+
+        public int IndexOf(string item)
+        {
+            _cmd.CommandText = $"SELECT `index` FROM {_tableName} WHERE name = '{item}'";
+            return Convert.ToInt32(_cmd.ExecuteScalar());
+        }
+
+        public void Insert(int index, string item)
+        {
+            _cmd.CommandText = $"UPDATE {_tableName} SET `index` = `index` + 1 WHERE `index` >= {index}";
+            _cmd.ExecuteNonQuery();
+            _cmd.CommandText = $"INSERT INTO {_tableName}(name, `index`) VALUES ('{item}', {index})";
+            _cmd.ExecuteNonQuery();
+            Count += 1;
+        }
+
+        public void RemoveAt(int index)
+        {
+            _cmd.CommandText = $"DELETE FROM {_tableName} WHERE `index` = {index}";
+            _cmd.ExecuteNonQuery();
+            _cmd.CommandText = $"UPDATE {_tableName} SET `index` = `index` - 1 WHERE `index` > {index}";
+            _cmd.ExecuteNonQuery();
+            Count -= 1;
+        }
+
+        public string this[int index]
+        {
+            get
+            {
+                _cmd.CommandText = $"SELECT name FROM {_tableName} WHERE `index` = {index}";
+                return (string)_cmd.ExecuteScalar();
+            }
+            set
+            {
+                _cmd.CommandText = $"UPDATE {_tableName} SET name = '{value}' WHERE `index` = {index}";

```

Comment:
I would use methods to select or update data instead of indexers because indexers are usually used for encapsulating an internal collection or array from  [MSDN](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/using-indexers)
>  Indexers are most frequently implemented in types whose primary purpose is to encapsulate an internal collection or array

Reply:
I think sharedstirngs is a duck `IList<T>` type. it is used like a collection or array, but the data is stored locally

Another reason is that the previous implementation of sharedstrings used List<string>, so the changes were minimal

Reply:
Did you mean dataBase is equal to collection from your option?

Reply:
Yes, I use sqlite just because it's easy to use, and I'm not find other appropriate file cache/disk cache.
if there is better lib, we can replace sqlite